### PR TITLE
Wasm docs

### PR DESCRIPTION
**Pull request by EItanya**

### CODE REVIEW

File: docs/content/advanced_configuration/wasm.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+description: Using WASM filters in Envoy with Gloo
+---
+
+# Envoy WASM
+
+Experimental support for Envoy WASM filters has been added to Open Source Gloo as of version 1.2.6.
+
+{{% notice note %}}
+This feature is considered incredibly experimental. It uses a fork of Envoy which does not have any official support. It 
+is meant to show off the potential of WASM filtes, and how they will integrate with Gloo going forward.
```
Comment (kdorosh) [inline docs/content/advanced_configuration/wasm.md]: typo: filtes

File: docs/content/advanced_configuration/wasm.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+Getting started with Envoy is simple, it requires setting one new value in the gloo helm chart.
+
+The easiest way to install with this value is using helm 3 as follows:
+{{< tabs >}}
+{{< tab name="helm 3" codelang="shell script">}}
+helm repo add gloo https://storage.googleapis.com/solo-public-helm
+
+helm repo update
+
+helm install --namespace gloo-system --set global.wasm.enabled=true gloo gloo/gloo
```
Comment (kdorosh) [inline docs/content/advanced_configuration/wasm.md]: let's include commands for helm 2 & helm 3, as users will be on both for the foreseeable future

File: docs/content/advanced_configuration/wasm.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+{{% notice note %}}
+This feature is considered incredibly experimental. It uses a fork of Envoy which does not have any official support. It 
+is meant to show off the potential of WASM filtes, and how they will integrate with Gloo going forward.
+{{% /notice %}}
+
+
+### Configuration
+
+Getting started with Envoy is simple, it requires setting one new value in the gloo helm chart.
```
Comment (christian-posta) [inline docs/content/advanced_configuration/wasm.md]: Getting started with WASM

File: docs/content/advanced_configuration/wasm.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+NAME                            READY   STATUS    RESTARTS   AGE
+discovery-5ff9ddbc8f-p7njb      1/1     Running   0          45s
+gateway-578f5b7d9d-khw8m        1/1     Running   0          45s
+gateway-proxy-c9b4cc476-x6h5j   1/1     Running   0          45s
+gloo-6889d56b5c-f28gv           1/1     Running   0          45s
+``` 
+
+Once all of the pods are up and running you are all ready to configure your first WASM filter.
+The API to configure the filter can be found [here]({{< ref "/api/github.com/solo-io/gloo/projects/gloo/api/v1/options/wasm/wasm.proto.sk.md" >}}).
```
Comment (christian-posta) [inline docs/content/advanced_configuration/wasm.md]: Where is this API doc?

File: docs/content/advanced_configuration/wasm.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        image: webassemblyhub.io/yuval-k/metrics:v1
+        name: yuval
+        root_id: stats_root_id
+```
+
+Once that is saved, the hard work has been done. All traffic on the http gateway will call the wasm filter.
+
+To find our more information about WASM filters, and how to build/run them check out [`wasme`](https://github.com/solo-io/wasme).
+
+`wasme` is a tool for building and deploying Envoy WASM filters, in Gloo, and in vanilla Envoy. Much more detailed information
```
Comment (christian-posta) [inline docs/content/advanced_configuration/wasm.md]: Can we mention something about WebAssembly Hub as well?

File: docs/content/advanced_configuration/wasm.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+At the moment the config must live on the gateway level, this will change as the Envoy WASM api evolves. To configure a gateway
+to add a WASM filter, the gateway must be edited like so.
+
+`kubectl edit -n gloo-system gateways.gateway.solo.io gateway-proxy`
+
+```yaml
+  httpGateway:
+    options:
+      wasm:
```
Comment (christian-posta) [inline docs/content/advanced_configuration/wasm.md]: Just double checking: we don't need `extensions` here? as in `httpGateway.options.extensions.config.wasm`?

Comment (christian-posta) [reply docs/content/advanced_configuration/wasm.md]: NM.. just verified the change locally

File: docs/content/advanced_configuration/wasm.md (hunk 4 above)
Comment (EItanya) [reply docs/content/advanced_configuration/wasm.md]: in the docs/api folder, under that path

File: docs/content/advanced_configuration/wasm.md (hunk 2 above)
Comment (EItanya) [reply docs/content/advanced_configuration/wasm.md]: Ok, in this case they are actually the same, so I will rename to just helm

File: docs/content/advanced_configuration/wasm.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+To check run the following:
+```shell script
+$ kubectl get pods -n gloo-system
+
+NAME                            READY   STATUS    RESTARTS   AGE
+discovery-5ff9ddbc8f-p7njb      1/1     Running   0          45s
+gateway-578f5b7d9d-khw8m        1/1     Running   0          45s
+gateway-proxy-c9b4cc476-x6h5j   1/1     Running   0          45s
+gloo-6889d56b5c-f28gv           1/1     Running   0          45s
+```
```
Comment (kdorosh) [inline docs/content/advanced_configuration/wasm.md]: can we break this into two code blocks? so the `kubectl` command is copy-pasteable

**Pull request merged**